src/test/java/com/assignmentTest/Assignment18Test.java

```diff
@@ -0,0 +1,26 @@
+package com.assignmentTest;
+
+import com.assignment.*;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+public class Assignment18Test {
+
+    @Test
+    public void testStudent() {
+        String[] infoBarrack = {"1", "Barrack Obama", "White House", "10", "PMMIWS"};
+        String[] infoJoe = {"2", "Joe Biden", "New White House", "2", "MES"};
+        String[] infoNewBarrack = {"1", "Barrack Obama", "Tudor House", "10", "PMMIWS"};
+        ArrayList<Student> studentArray = new ArrayList<>();
```

Comment:
Creating this arraylist here leads to more overhead. Lets modify this design by removing the creation here. 
Rather create this in `StudentInfo` class as a global variable. By doing this you won't even need to pass this for all your methods.